commit 072f06abf713d7536ce2a84a5ac266f585c88d39
Author: Matthias Clasen <mclasen@redhat.com>
Date:   Fri Dec 8 17:48:47 2017 -0500

    Fix fallout from g_object_ref change

    g_object_ref now returns the type of the object that was
    passed. Introduce cast as necessary to avoid warnings due
    to this.

Index: gtk-2.24.33/gtk/gtkcellrendererspin.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkcellrendererspin.c
+++ gtk-2.24.33/gtk/gtkcellrendererspin.c
@@ -207,7 +207,8 @@ gtk_cell_renderer_spin_set_property (GOb
 	}
 
       if (obj)
-	priv->adjustment = g_object_ref_sink (obj);
+	priv->adjustment = GTK_ADJUSTMENT (g_object_ref_sink (obj));
+
       break;
     case PROP_CLIMB_RATE:
       priv->climb_rate = g_value_get_double (value);
Index: gtk-2.24.33/gtk/gtkcomboboxtext.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkcomboboxtext.c
+++ gtk-2.24.33/gtk/gtkcomboboxtext.c
@@ -275,7 +275,7 @@ gtk_combo_box_text_buildable_custom_tag_
 
       parser_data = g_slice_new0 (ItemParserData);
       parser_data->builder = g_object_ref (builder);
-      parser_data->object = g_object_ref (buildable);
+      parser_data->object = (GObject *)g_object_ref (buildable);
       parser_data->domain = gtk_builder_get_translation_domain (builder);
       *parser = item_parser;
       *data = parser_data;
Index: gtk-2.24.33/gtk/gtklabel.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtklabel.c
+++ gtk-2.24.33/gtk/gtklabel.c
@@ -1336,7 +1336,7 @@ gtk_label_buildable_custom_tag_start (Gt
 
       parser_data = g_slice_new0 (PangoParserData);
       parser_data->builder = g_object_ref (builder);
-      parser_data->object = g_object_ref (buildable);
+      parser_data->object = (GObject *) g_object_ref (buildable);
       *parser = pango_parser;
       *data = parser_data;
       return TRUE;
Index: gtk-2.24.33/gtk/gtktoolpalette.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtktoolpalette.c
+++ gtk-2.24.33/gtk/gtktoolpalette.c
@@ -734,7 +734,7 @@ gtk_tool_palette_add (GtkContainer *cont
 
   g_ptr_array_add (palette->priv->groups, info);
   info->pos = palette->priv->groups->len - 1;
-  info->widget = g_object_ref_sink (child);
+  info->widget = (GtkToolItemGroup *)g_object_ref_sink (child);
 
   gtk_widget_set_parent (child, GTK_WIDGET (palette));
 }
Index: gtk-2.24.33/gtk/gtktreeview.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtktreeview.c
+++ gtk-2.24.33/gtk/gtktreeview.c
@@ -14147,7 +14147,7 @@ gtk_tree_view_set_search_entry (GtkTreeV
 
   if (entry)
     {
-      tree_view->priv->search_entry = g_object_ref (entry);
+      tree_view->priv->search_entry = GTK_WIDGET (g_object_ref (entry));
       tree_view->priv->search_custom_entry_set = TRUE;
 
       if (tree_view->priv->search_entry_changed_id == 0)
Index: gtk-2.24.33/gtk/gtkwidget.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkwidget.c
+++ gtk-2.24.33/gtk/gtkwidget.c
@@ -10679,7 +10679,7 @@ gtk_widget_buildable_custom_tag_start (G
       AccelGroupParserData *parser_data;
 
       parser_data = g_slice_new0 (AccelGroupParserData);
-      parser_data->object = g_object_ref (buildable);
+      parser_data->object = (GObject *)g_object_ref (buildable);
       *parser = accel_group_parser;
       *data = parser_data;
       return TRUE;
Index: gtk-2.24.33/modules/printbackends/cups/gtkprintbackendcups.c
===================================================================
--- gtk-2.24.33.orig/modules/printbackends/cups/gtkprintbackendcups.c
+++ gtk-2.24.33/modules/printbackends/cups/gtkprintbackendcups.c
@@ -3609,7 +3609,7 @@ cups_request_ppd (GtkPrinter *printer)
   g_io_channel_set_encoding (data->ppd_io, NULL, NULL);
   g_io_channel_set_close_on_unref (data->ppd_io, TRUE);
 
-  data->printer = g_object_ref (printer);
+  data->printer = (GtkPrinterCups *) g_object_ref (printer);
 
   resource = g_strdup_printf ("/printers/%s.ppd", 
                               gtk_printer_cups_get_ppd_name (GTK_PRINTER_CUPS (printer)));
Index: gtk-2.24.33/tests/testentrycompletion.c
===================================================================
--- gtk-2.24.33.orig/tests/testentrycompletion.c
+++ gtk-2.24.33/tests/testentrycompletion.c
@@ -237,7 +237,7 @@ animation_timer (GtkEntryCompletion *com
 	{
 	  g_print ("removing model!\n");
 
-	  old_store = g_object_ref (gtk_entry_completion_get_model (completion));
+	  old_store = GTK_LIST_STORE (g_object_ref (gtk_entry_completion_get_model (completion)));
 	  gtk_entry_completion_set_model (completion, NULL);
 	}
       else
Index: gtk-2.24.33/gtk/gtkimmodule.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkimmodule.c
+++ gtk-2.24.33/gtk/gtkimmodule.c
@@ -662,7 +662,7 @@ lookup_immodule (gchar **immodules_list)
 	  gboolean found;
 	  gchar *context_id;
 	  found = g_hash_table_lookup_extended (contexts_hash, *immodules_list,
-						&context_id, NULL);
+						(gpointer *) &context_id, NULL);
 	  if (found)
 	    return context_id;
 	}
Index: gtk-2.24.33/gtk/gtkscale.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkscale.c
+++ gtk-2.24.33/gtk/gtkscale.c
@@ -1471,7 +1471,7 @@ gtk_scale_add_mark (GtkScale        *sca
   mark->position = position;
  
   priv->marks = g_slist_insert_sorted_with_data (priv->marks, mark,
-                                                 (GCompareFunc) compare_marks,
+                                                 compare_marks,
                                                  GINT_TO_POINTER (
                                                    gtk_range_get_inverted (GTK_RANGE (scale)) 
                                                    ));
Index: gtk-2.24.33/gtk/gtkuimanager.c
===================================================================
--- gtk-2.24.33.orig/gtk/gtkuimanager.c
+++ gtk-2.24.33/gtk/gtkuimanager.c
@@ -503,7 +503,7 @@ gtk_ui_manager_buildable_construct_child
   g_signal_connect (widget, "hierarchy-changed",
 		    G_CALLBACK (child_hierarchy_changed_cb),
 		    GTK_UI_MANAGER (buildable));
-  return g_object_ref (widget);
+  return G_OBJECT (g_object_ref (widget));
 }
 
 static void
Index: gtk-2.24.33/gtk/tests/defaultvalue.c
===================================================================
--- gtk-2.24.33.orig/gtk/tests/defaultvalue.c
+++ gtk-2.24.33/gtk/tests/defaultvalue.c
@@ -91,13 +91,13 @@ test_type (gconstpointer data)
   klass = g_type_class_ref (type);
   
   if (g_type_is_a (type, GTK_TYPE_SETTINGS))
-    instance = g_object_ref (gtk_settings_get_default ());
+    instance = G_OBJECT (g_object_ref (gtk_settings_get_default ()));
   else if (g_type_is_a (type, GDK_TYPE_PANGO_RENDERER))
-    instance = g_object_ref (gdk_pango_renderer_get_default (gdk_screen_get_default ()));
+    instance = G_OBJECT (g_object_ref (gdk_pango_renderer_get_default (gdk_screen_get_default ())));
   else if (g_type_is_a (type, GDK_TYPE_PIXMAP))
-    instance = g_object_ref (gdk_pixmap_new (NULL, 1, 1, 1));
+    instance = G_OBJECT (g_object_ref (gdk_pixmap_new (NULL, 1, 1, 1)));
   else if (g_type_is_a (type, GDK_TYPE_COLORMAP))
-    instance = g_object_ref (gdk_colormap_new (gdk_visual_get_best (), TRUE));
+    instance = G_OBJECT (g_object_ref (gdk_colormap_new (gdk_visual_get_best (), TRUE)));
   else if (g_type_is_a (type, GDK_TYPE_WINDOW))
     {
       GdkWindowAttr attributes;
@@ -105,7 +105,7 @@ test_type (gconstpointer data)
       attributes.event_mask = 0;
       attributes.width = 100;
       attributes.height = 100;
-      instance = g_object_ref (gdk_window_new (NULL, &attributes, 0));
+      instance = G_OBJECT (g_object_ref (gdk_window_new (NULL, &attributes, 0)));
     }
   else
     instance = g_object_new (type, NULL);
Index: gtk-2.24.33/modules/other/gail/gailtreeview.c
===================================================================
--- gtk-2.24.33.orig/modules/other/gail/gailtreeview.c
+++ gtk-2.24.33/modules/other/gail/gailtreeview.c
@@ -995,7 +995,7 @@ gail_tree_view_ref_child (AtkObject *obj
         gail_cell_add_state (cell, ATK_STATE_FOCUSABLE, FALSE);
         if (focus_index == i)
           {
-            gailview->focus_cell = g_object_ref (cell);
+            gailview->focus_cell = ATK_OBJECT (g_object_ref (cell));
             gail_cell_add_state (cell, ATK_STATE_FOCUSED, FALSE);
             g_signal_emit_by_name (gailview,
                                    "active-descendant-changed",
Index: gtk-2.24.33/tests/testmenubars.c
===================================================================
--- gtk-2.24.33.orig/tests/testmenubars.c
+++ gtk-2.24.33/tests/testmenubars.c
@@ -21,7 +21,7 @@
 #include <gtk/gtk.h>
 
 static GtkWidget *
-create_menu (depth)
+create_menu (guint depth)
 {
     GtkWidget *menu;
     GtkWidget *menuitem;
@@ -70,19 +70,19 @@ create_menubar (GtkPackDirection pack_di
   menuitem = gtk_image_menu_item_new_from_stock (GTK_STOCK_HOME, NULL);
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   menuitem = gtk_menu_item_new_with_label ("foo");
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   menuitem = gtk_menu_item_new_with_label ("bar");
   gtk_menu_shell_append (GTK_MENU_SHELL (menubar), menuitem);
   gtk_label_set_angle (GTK_LABEL (GTK_BIN (menuitem)->child), angle);
-  menu = create_menu (2, TRUE);
+  menu = create_menu (2);
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
 
   return menubar;
